src/rocjpeg_vaapi_decoder.cpp

```diff
@@ -778,10 +802,25 @@ RocJpegStatus RocJpegVappiDecoder::SubmitDecodeBatched(JpegStreamParameters *jpe
         surface_format = key.surface_format;
         surface_attrib.value.value.i = key.pixel_format;
 
+        std::vector<VASurfaceAttrib> surface_attribs;
+        surface_attribs.push_back(surface_attrib);
+
+        uint64_t mod_linear = 0;
+        VADRMFormatModifierList modifier_list = {
+            .num_modifiers = 1,
+            .modifiers = &mod_linear,
+        };
+        if (supports_modifiers_) {
+            surface_attrib.type = VASurfaceAttribDRMFormatModifiers;
+            surface_attrib.value.type = VAGenericValueTypePointer;
+            surface_attrib.value.value.p = &modifier_list;
+            surface_attribs.push_back(surface_attrib);
+        }
+
         RocJpegVaapiMemPoolEntry mem_pool_entry = vaapi_mem_pool_->GetEntry(key.pixel_format, key.width, key.height, indices.size());
         if (mem_pool_entry.va_surface_ids.empty()) {
             mem_pool_entry.va_surface_ids.resize(indices.size());
-            CHECK_VAAPI(vaCreateSurfaces(va_display_, surface_format, key.width, key.height, mem_pool_entry.va_surface_ids.data(), mem_pool_entry.va_surface_ids.size(), &surface_attrib, 1));
+            CHECK_VAAPI(vaCreateSurfaces(va_display_, surface_format, key.width, key.height, mem_pool_entry.va_surface_ids.data(), mem_pool_entry.va_surface_ids.size(), surface_attribs.data(), surface_attribs.size()));
```

Comment:
Which driver version will have Mesa using tiling by default, and for which ASICs? This change needs further internal discussion and testing for clarity.